src/api/DWGraph_DS.java

```diff
@@ -74,11 +74,17 @@ public void addNode(node_data n) {
      */
     @Override
     public void connect(int src, int dest, double w) {
-        if (w > 0) {
-            outEdges.get(src).put(dest, new EdgeData(src, dest, w)); //Put a new outgoing edge from src to dest
-            inEdges.get(dest).put(src, new EdgeData(src, dest, w)); //Put a new incoming edge from dest to src
-            edgeSize++; //A new edge is added - increment edgeSize
-            countMC++; //Count 1 meta-change.
+        if (nodes.containsKey(src) && nodes.containsKey(dest)) {
+            if (w >= 0) {
+                if (!nodes.containsKey(src) && !nodes.containsKey(dest)) {
+                    edgeSize++; //A new edge is added - increment edgeSize
+                    countMC++; //Count 1 meta-change.
```

Comment:
Why we check if NOT nodes.containsKey... and then increment the edgeSize?
I think last implementation is good with the new edge cases.